## Multilayer Perceptrons Model (2-D tensor / Matrix)
- It is generally created for 2-dimensional data

In [1]:
## Importing necessary lib
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline
import seaborn as sns
import plotly.express as px
import warnings
warnings.filterwarnings("ignore")
plt.style.use('dark_background')

from sklearn.model_selection import train_test_split

from numpy import loadtxt
from tensorflow import keras 
from keras.models import Sequential
from keras.layers import Dense, Dropout
from keras import initializers

### 18.2.1 Basic Model (Model_1):

In [2]:
## Step - 1 : Data Preparation
credit_card = pd.read_csv("D:\Datasets\creditcard_BM_DL.csv")

In [3]:
credit_card.shape

(284807, 31)

In [4]:
credit_card.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284807 entries, 0 to 284806
Data columns (total 31 columns):
 #   Column  Non-Null Count   Dtype  
---  ------  --------------   -----  
 0   Time    284807 non-null  float64
 1   V1      284807 non-null  float64
 2   V2      284807 non-null  float64
 3   V3      284807 non-null  float64
 4   V4      284807 non-null  float64
 5   V5      284807 non-null  float64
 6   V6      284807 non-null  float64
 7   V7      284807 non-null  float64
 8   V8      284807 non-null  float64
 9   V9      284807 non-null  float64
 10  V10     284807 non-null  float64
 11  V11     284807 non-null  float64
 12  V12     284807 non-null  float64
 13  V13     284807 non-null  float64
 14  V14     284807 non-null  float64
 15  V15     284807 non-null  float64
 16  V16     284807 non-null  float64
 17  V17     284807 non-null  float64
 18  V18     284807 non-null  float64
 19  V19     284807 non-null  float64
 20  V20     284807 non-null  float64
 21  V21     28

In [5]:
# credit_card = credit_card.drop(['Time'], axis = 1)

In [6]:
# credit_card.shape

In [7]:
# credit_card.info()

In [8]:
# Split into input (X) and output (y) variables
X = credit_card.iloc[:,0:30]
y = credit_card.iloc[:,30]

In [9]:
## Data splitting
np.random.seed(500)
x_train, x_test, y_train, y_test = train_test_split(X, y, test_size = 0.3, random_state = 0)
print('Shape of training dataset is :',x_train.shape)
print('')
print('Shape of test dataset is :',x_test.shape)

Shape of training dataset is : (199364, 30)

Shape of test dataset is : (85443, 30)


In [10]:
## Determining the number of columns
input_dim = x_train.shape[1]

In [11]:
## Step- 2 : Creating First model
print('------------------ MODEL_1 ------------------')

# Creating an empty sequential model
model_1 = Sequential()

# Adding an input layer to the model i.e Layer 1
model_1.add(Dense(10, input_dim = input_dim, activation = 'relu', kernel_initializer = 'uniform'))

# Adding a hidden layer(here drop out layer but without any drop out)
model_1.add(Dropout(0))

# Adding an output layer to the model
model_1.add(Dense(1, kernel_initializer = 'uniform', activation = 'softmax'))

------------------ MODEL_1 ------------------


In [12]:
model_1.summary()

Model: "sequential"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
 dense (Dense)               (None, 10)                310       
                                                                 
 dropout (Dropout)           (None, 10)                0         
                                                                 
 dense_1 (Dense)             (None, 1)                 11        
                                                                 
Total params: 321
Trainable params: 321
Non-trainable params: 0
_________________________________________________________________


In [13]:
## Step- 3 : Compile the model
model_1.compile(loss = 'binary_crossentropy', optimizer = 'SGD', metrics = ['accuracy'])

In [14]:
## Step- 4 : Fit the model
model_1.fit(x_train, y_train, epochs = 5, batch_size = 1000)

Epoch 1/5
200/200 [==============================] - 1s 1ms/step - loss: 719021.2500 - accuracy: 0.0017
Epoch 2/5
200/200 [==============================] - 0s 1ms/step - loss: 0.3019 - accuracy: 0.0017
Epoch 3/5
200/200 [==============================] - 0s 1ms/step - loss: 0.2065 - accuracy: 0.0017
Epoch 4/5
200/200 [==============================] - 0s 1ms/step - loss: 0.1552 - accuracy: 0.0017
Epoch 5/5
200/200 [==============================] - 0s 1ms/step - loss: 0.1239 - accuracy: 0.0017


In [15]:
## Step- 5 : Evaluate the model
score = model_1.evaluate(x_train, y_train)
# print('Training Accuracy is : %.2f%%' %(100*score[1]))

6231/6231 [==============================] - 5s 797us/step - loss: 0.1121 - accuracy: 0.0017


In [16]:
score

[0.11214865744113922, 0.001730503048747778]

In [17]:
## Step- 5 : Evaluate the model
score_trn = model_1.evaluate(x_train, y_train)
print('Training Accuracy is : %.2f%%' %(100*score_trn[1]))
print('')

score_tst = model_1.evaluate(x_test, y_test)
print('Test Accuracy is : %.2f%%' %(100*score_tst[1]))
print('')

6231/6231 [==============================] - 5s 753us/step - loss: 0.1121 - accuracy: 0.0017
Training Accuracy is : 0.17%

2671/2671 [==============================] - 2s 795us/step - loss: 0.1121 - accuracy: 0.0017
Test Accuracy is : 0.17%



#### Obs : Model accuracy is very poor/less i.e. 0.17%

### 18.2.2 Model Developement (Model_2) : Changing Units, Dropout, Epoch and Batch_size

In [18]:
## Step- 2 : Creating First model
print('------------------ MODEL_2 ------------------')

# Creating an empty sequential model
model_2 = Sequential()

# Adding an input layer to the model i.e Layer 1 [unit : 10 to 1000]
model_2.add(Dense(1000, input_dim = input_dim, activation = 'relu', kernel_initializer = 'uniform'))

# Adding a hidden layer  [drop out : 0 to 0.1]
model_2.add(Dropout(0.1))

# Adding an output layer to the model
model_2.add(Dense(1, kernel_initializer = 'uniform', activation = 'softmax'))

## Step- 3 : Compile the model
model_2.compile(loss = 'binary_crossentropy', optimizer = 'SGD', metrics = ['accuracy'])

## Step- 4 : Fit the model [epochs: 5 to 500( to avoid time I am taking 50), batch_size: 1000 to 2000]
model_2.fit(x_train, y_train, epochs = 50, batch_size = 2000)

------------------ MODEL_2 ------------------
Epoch 1/50
100/100 [==============================] - 4s 37ms/step - loss: 2779528560640.0000 - accuracy: 0.0017
Epoch 2/50
100/100 [==============================] - 4s 37ms/step - loss: 227.7416 - accuracy: 0.0017
Epoch 3/50
100/100 [==============================] - 4s 37ms/step - loss: 8719.5098 - accuracy: 0.0017
Epoch 4/50
100/100 [==============================] - 4s 36ms/step - loss: 7.5414 - accuracy: 0.0017
Epoch 5/50
100/100 [==============================] - 4s 36ms/step - loss: 1763.7444 - accuracy: 0.0017
Epoch 6/50
100/100 [==============================] - 4s 36ms/step - loss: 1572.1207 - accuracy: 0.0017
Epoch 7/50
100/100 [==============================] - 4s 38ms/step - loss: 4403.8867 - accuracy: 0.0017
Epoch 8/50
100/100 [==============================] - 4s 36ms/step - loss: 2.0018 - accuracy: 0.0017
Epoch 9/50
100/100 [==============================] - 4s 36ms/step - loss: 0.1290 - accuracy: 0.0017
Epoch 10/50
100/100

In [19]:
## Step- 5 : Evaluate the model
score_trn = model_2.evaluate(x_train, y_train)
print('Training Accuracy is : %.2f%%' %(100*score_trn[1]))
print('')

score_tst = model_2.evaluate(x_test, y_test)
print('Test Accuracy is : %.2f%%' %(100*score_tst[1]))
print('')

6231/6231 [==============================] - 5s 783us/step - loss: 0.0282 - accuracy: 0.0017
Training Accuracy is : 0.17%

2671/2671 [==============================] - 2s 875us/step - loss: 0.0282 - accuracy: 0.0017
Test Accuracy is : 0.17%



#### Obs : Still there is no improvement of accuracy i.e accuracy is 0.17%

### 18.2.3 Model Developement (Model_3) : Changing Activation, Loss and Optimizer

In [20]:
## Step- 2 : Creating First model
print('------------------ MODEL_3 ------------------')

# Creating an empty sequential model
model_3 = Sequential()

# Adding an input layer to the model i.e Layer 1 [unit : 10 to 1000 to 100] [act Relu to softmax]
model_3.add(Dense(100, input_dim = input_dim, activation = 'softmax', kernel_initializer = 'uniform'))

# Adding a hidden layer  [drop out : 0 to 0.1]
model_3.add(Dropout(0.1))

# Adding an output layer to the model  [softmax to sigmoid]
model_3.add(Dense(1, kernel_initializer = 'uniform', activation = 'sigmoid'))

## Step- 3 : Compile the model [opt: SGD to RMSprop ]
model_3.compile(loss = 'binary_crossentropy', optimizer = 'RMSprop', metrics = ['accuracy'])

## Step- 4 : Fit the model [epochs: 5 to 500 to 50, batch_size: 1000 to 2000 to 2000]
model_3.fit(x_train, y_train, epochs = 50, batch_size = 2000)

------------------ MODEL_3 ------------------
Epoch 1/50
100/100 [==============================] - 1s 5ms/step - loss: 0.6127 - accuracy: 0.9982
Epoch 2/50
100/100 [==============================] - 0s 5ms/step - loss: 0.5269 - accuracy: 0.9983
Epoch 3/50
100/100 [==============================] - 0s 5ms/step - loss: 0.4516 - accuracy: 0.9983
Epoch 4/50
100/100 [==============================] - 0s 5ms/step - loss: 0.3858 - accuracy: 0.9983
Epoch 5/50
100/100 [==============================] - 0s 5ms/step - loss: 0.3285 - accuracy: 0.9983
Epoch 6/50
100/100 [==============================] - 0s 5ms/step - loss: 0.2789 - accuracy: 0.9983
Epoch 7/50
100/100 [==============================] - 0s 5ms/step - loss: 0.2365 - accuracy: 0.9983
Epoch 8/50
100/100 [==============================] - 0s 5ms/step - loss: 0.2005 - accuracy: 0.9983
Epoch 9/50
100/100 [==============================] - 0s 5ms/step - loss: 0.1703 - accuracy: 0.9983
Epoch 10/50
100/100 [==============================] -

In [21]:
## Step- 5 : Evaluate the model
score_trn = model_3.evaluate(x_train, y_train)
print('Training Accuracy is : %.2f%%' %(100*score_trn[1]))
print('')

score_tst = model_3.evaluate(x_test, y_test)
print('Test Accuracy is : %.2f%%' %(100*score_tst[1]))
print('')

6231/6231 [==============================] - 5s 835us/step - loss: 0.0128 - accuracy: 0.9983
Training Accuracy is : 99.83%

2671/2671 [==============================] - 2s 835us/step - loss: 0.0127 - accuracy: 0.9983
Test Accuracy is : 99.83%



#### Obs : Excellent Result. Accuracy increased to 99.83% for both train and test which means there is no overfitting and underfitting

### 18.2.4 Model Developement (Model_4) : Changing Activation & Optimizer

In [22]:
## Step- 2 : Creating First model
print('------------------ MODEL_4 ------------------')

# Creating an empty sequential model
model_4 = Sequential()

# Adding an input layer to the model i.e Layer 1 [unit : 10 to 1000 to 100] [act Relu to softmax to relu]
model_4.add(Dense(1000, input_dim = input_dim, activation = 'relu', kernel_initializer = 'uniform'))

# Adding a hidden layer  [drop out : 0 to 0.1]
model_4.add(Dropout(0.1))

# Adding an output layer to the model  [softmax to sigmoid]
model_4.add(Dense(1, kernel_initializer = 'uniform', activation = 'sigmoid'))

## Step- 3 : Compile the model [opt: SGD to RMSprop to adagrad ]
model_4.compile(loss = 'binary_crossentropy', optimizer = 'adagrad', metrics = ['accuracy'])

## Step- 4 : Fit the model [epochs: 5 to 500 to 50, batch_size: 1000 to 2000 to 2000]
model_4.fit(x_train, y_train, epochs = 50, batch_size = 2000)

------------------ MODEL_4 ------------------
Epoch 1/50
100/100 [==============================] - 4s 36ms/step - loss: 3.7150 - accuracy: 0.9957
Epoch 2/50
100/100 [==============================] - 4s 36ms/step - loss: 2.7024 - accuracy: 0.9967
Epoch 3/50
100/100 [==============================] - 4s 36ms/step - loss: 2.7432 - accuracy: 0.9968
Epoch 4/50
100/100 [==============================] - 4s 36ms/step - loss: 2.9236 - accuracy: 0.9967
Epoch 5/50
100/100 [==============================] - 4s 36ms/step - loss: 2.8590 - accuracy: 0.9968
Epoch 6/50
100/100 [==============================] - 4s 36ms/step - loss: 2.7302 - accuracy: 0.9967
Epoch 7/50
100/100 [==============================] - 4s 36ms/step - loss: 2.8329 - accuracy: 0.9968
Epoch 8/50
100/100 [==============================] - 4s 36ms/step - loss: 2.7844 - accuracy: 0.9967
Epoch 9/50
100/100 [==============================] - 4s 37ms/step - loss: 2.6639 - accuracy: 0.9967
Epoch 10/50
100/100 [========================

In [23]:
## Step- 5 : Evaluate the model
score_trn = model_4.evaluate(x_train, y_train)
print('Training Accuracy is : %.2f%%' %(100*score_trn[1]))
print('')

score_tst = model_4.evaluate(x_test, y_test)
print('Test Accuracy is : %.2f%%' %(100*score_tst[1]))
print('')

6231/6231 [==============================] - 6s 887us/step - loss: 2.0153 - accuracy: 0.9983
Training Accuracy is : 99.83%

2671/2671 [==============================] - 3s 950us/step - loss: 2.0555 - accuracy: 0.9983
Test Accuracy is : 99.83%



### 18.2.5 Hyper parameter - Grid Approach to Determine Best value of Epoch and Batch Size
- To improve the accuracy we need to determine the best value of epoch and best size

In [27]:
## Step- 2 : Creating Fifth model by using Grid search
from keras.constraints import maxnorm
print('------------------ MODEL_5 ------------------')

# Values of weight_constraints can be used are : [0,1,2,3,4,5]
weight_constraint = 0

# Creating a function
def create_model():
    model = Sequential()
    
    # Using maxnorm constraints
    model.add(Dense(1000, input_dim = input_dim, activation = 'relu', kernel_initializer = 'uniform',
                   kernel_constraint = maxnorm(weight_constraint)))
    model.add(Dropout(0.1))
    model.add(Dense(1, kernel_initializer = 'uniform', activation = 'sigmoid'))
    
    # Compile model
    model.compile(loss = 'binary_crossentropy', optimizer = 'adagrad', metrics = ['accuracy'])
    
    # Return the model
    return model

## Create model using the above defined function
from keras.wrappers.scikit_learn import KerasClassifier
model_5 = KerasClassifier(build_fn = create_model)

## Creating a Grid of different Epochs and batch_size
from sklearn.model_selection import GridSearchCV
epochs = [50,100]
batch_size = [1500,2500,3000]

## Creating a dictionary
pram_grid = dict(epochs = epochs, batch_size = batch_size)

## Performing Grid search w.r.t epochs and batch_size
grid = GridSearchCV(estimator = model_5, param_grid = pram_grid)
grid_result = grid.fit(x_train, y_train)

## Display the complete results
print('The Complete Results: \n',grid_result.cv_results_)
print('')

# Display the summarize results
print('Best: %f using %s' % (grid_result.best_score_, grid_result.best_params_))

------------------ MODEL_5 ------------------
Epoch 1/50
107/107 [==============================] - 4s 30ms/step - loss: 7.3557 - accuracy: 0.9900
Epoch 2/50
107/107 [==============================] - 3s 29ms/step - loss: 0.6526 - accuracy: 0.9983
Epoch 3/50
107/107 [==============================] - 3s 29ms/step - loss: 0.6325 - accuracy: 0.9983
Epoch 4/50
107/107 [==============================] - 3s 29ms/step - loss: 0.6143 - accuracy: 0.9983
Epoch 5/50
107/107 [==============================] - 3s 29ms/step - loss: 0.5973 - accuracy: 0.9983
Epoch 6/50
107/107 [==============================] - 3s 29ms/step - loss: 0.5810 - accuracy: 0.9983
Epoch 7/50
107/107 [==============================] - 3s 29ms/step - loss: 0.5651 - accuracy: 0.9983
Epoch 8/50
107/107 [==============================] - 3s 32ms/step - loss: 0.5494 - accuracy: 0.9983
Epoch 9/50
107/107 [==============================] - 4s 42ms/step - loss: 0.5340 - accuracy: 0.9983
Epoch 10/50
107/107 [========================

107/107 [==============================] - 3s 32ms/step - loss: 0.2427 - accuracy: 0.9983
Epoch 31/50
107/107 [==============================] - 3s 32ms/step - loss: 0.2340 - accuracy: 0.9983
Epoch 32/50
107/107 [==============================] - 4s 36ms/step - loss: 0.2257 - accuracy: 0.9983
Epoch 33/50
107/107 [==============================] - 3s 32ms/step - loss: 0.2178 - accuracy: 0.9983
Epoch 34/50
107/107 [==============================] - 3s 32ms/step - loss: 0.2102 - accuracy: 0.9983
Epoch 35/50
107/107 [==============================] - 4s 33ms/step - loss: 0.2030 - accuracy: 0.9983
Epoch 36/50
107/107 [==============================] - 3s 32ms/step - loss: 0.1962 - accuracy: 0.9983
Epoch 37/50
107/107 [==============================] - 3s 31ms/step - loss: 0.1896 - accuracy: 0.9983
Epoch 38/50
107/107 [==============================] - 3s 31ms/step - loss: 0.1834 - accuracy: 0.9983
Epoch 39/50
107/107 [==============================] - 3s 32ms/step - loss: 0.1775 - accuracy:

107/107 [==============================] - 3s 32ms/step - loss: 0.6074 - accuracy: 0.9982
Epoch 9/50
107/107 [==============================] - 3s 33ms/step - loss: 0.5936 - accuracy: 0.9982
Epoch 10/50
107/107 [==============================] - 3s 33ms/step - loss: 0.5788 - accuracy: 0.9982
Epoch 11/50
107/107 [==============================] - 4s 33ms/step - loss: 0.5633 - accuracy: 0.9982
Epoch 12/50
107/107 [==============================] - 4s 33ms/step - loss: 0.5469 - accuracy: 0.9982
Epoch 13/50
107/107 [==============================] - 3s 32ms/step - loss: 0.5300 - accuracy: 0.9982
Epoch 14/50
107/107 [==============================] - 3s 32ms/step - loss: 0.5127 - accuracy: 0.9982
Epoch 15/50
107/107 [==============================] - 3s 32ms/step - loss: 0.4951 - accuracy: 0.9982
Epoch 16/50
107/107 [==============================] - 4s 33ms/step - loss: 0.4774 - accuracy: 0.9982
Epoch 17/50
107/107 [==============================] - 3s 32ms/step - loss: 0.4596 - accuracy: 

107/107 [==============================] - 4s 33ms/step - loss: 0.3584 - accuracy: 0.9983
Epoch 38/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3532 - accuracy: 0.9983
Epoch 39/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3481 - accuracy: 0.9983
Epoch 40/50
107/107 [==============================] - 4s 34ms/step - loss: 0.3432 - accuracy: 0.9983
Epoch 41/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3383 - accuracy: 0.9983
Epoch 42/50
107/107 [==============================] - 4s 34ms/step - loss: 0.3335 - accuracy: 0.9983
Epoch 43/50
107/107 [==============================] - 4s 35ms/step - loss: 0.3289 - accuracy: 0.9983
Epoch 44/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3243 - accuracy: 0.9983
Epoch 45/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3198 - accuracy: 0.9983
Epoch 46/50
107/107 [==============================] - 4s 33ms/step - loss: 0.3155 - accuracy:

107/107 [==============================] - 4s 35ms/step - loss: 0.1918 - accuracy: 0.9983
Epoch 67/100
107/107 [==============================] - 4s 37ms/step - loss: 0.1882 - accuracy: 0.9983
Epoch 68/100
107/107 [==============================] - 4s 37ms/step - loss: 0.1847 - accuracy: 0.9983
Epoch 69/100
107/107 [==============================] - 4s 36ms/step - loss: 0.1812 - accuracy: 0.9983
Epoch 70/100
107/107 [==============================] - 4s 35ms/step - loss: 0.1779 - accuracy: 0.9983
Epoch 71/100
107/107 [==============================] - 4s 36ms/step - loss: 0.1746 - accuracy: 0.9983
Epoch 72/100
107/107 [==============================] - 4s 34ms/step - loss: 0.1714 - accuracy: 0.9983
Epoch 73/100
107/107 [==============================] - 4s 35ms/step - loss: 0.1684 - accuracy: 0.9983
Epoch 74/100
107/107 [==============================] - 4s 35ms/step - loss: 0.1654 - accuracy: 0.9983
Epoch 75/100
107/107 [==============================] - 4s 37ms/step - loss: 0.1624 - 

107/107 [==============================] - 3s 29ms/step - loss: 0.2411 - accuracy: 0.9983
Epoch 45/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2357 - accuracy: 0.9983
Epoch 46/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2307 - accuracy: 0.9983
Epoch 47/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2257 - accuracy: 0.9983
Epoch 48/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2209 - accuracy: 0.9983
Epoch 49/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2162 - accuracy: 0.9983
Epoch 50/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2117 - accuracy: 0.9983
Epoch 51/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2073 - accuracy: 0.9983
Epoch 52/100
107/107 [==============================] - 3s 30ms/step - loss: 0.2031 - accuracy: 0.9983
Epoch 53/100
107/107 [==============================] - 3s 30ms/step - loss: 0.1989 - 

107/107 [==============================] - 3s 30ms/step - loss: 0.4031 - accuracy: 0.9982
Epoch 23/100
107/107 [==============================] - 3s 30ms/step - loss: 0.3932 - accuracy: 0.9982
Epoch 24/100
107/107 [==============================] - 3s 30ms/step - loss: 0.3835 - accuracy: 0.9982
Epoch 25/100
107/107 [==============================] - 3s 30ms/step - loss: 0.3741 - accuracy: 0.9982
Epoch 26/100
107/107 [==============================] - 3s 31ms/step - loss: 0.3649 - accuracy: 0.9982
Epoch 27/100
107/107 [==============================] - 3s 30ms/step - loss: 0.3558 - accuracy: 0.9982
Epoch 28/100
107/107 [==============================] - 3s 30ms/step - loss: 0.3471 - accuracy: 0.9982
Epoch 29/100
107/107 [==============================] - 3s 31ms/step - loss: 0.3385 - accuracy: 0.9982
Epoch 30/100
107/107 [==============================] - 3s 31ms/step - loss: 0.3301 - accuracy: 0.9982
Epoch 31/100
107/107 [==============================] - 3s 31ms/step - loss: 0.3219 - 

27/27 [==============================] - 0s 5ms/step - loss: 0.0851 - accuracy: 0.9985
Epoch 1/100
107/107 [==============================] - 4s 34ms/step - loss: 35.8973 - accuracy: 0.9889
Epoch 2/100
107/107 [==============================] - 4s 34ms/step - loss: 0.6568 - accuracy: 0.9982
Epoch 3/100
107/107 [==============================] - 4s 33ms/step - loss: 0.6388 - accuracy: 0.9982
Epoch 4/100
107/107 [==============================] - 4s 34ms/step - loss: 0.6227 - accuracy: 0.9982
Epoch 5/100
107/107 [==============================] - 3s 33ms/step - loss: 0.6076 - accuracy: 0.9982
Epoch 6/100
107/107 [==============================] - 3s 33ms/step - loss: 0.5931 - accuracy: 0.9982
Epoch 7/100
107/107 [==============================] - 4s 35ms/step - loss: 0.5790 - accuracy: 0.9982
Epoch 8/100
107/107 [==============================] - 4s 35ms/step - loss: 0.5652 - accuracy: 0.9982
Epoch 9/100
107/107 [==============================] - 4s 33ms/step - loss: 0.5515 - accuracy: 0

107/107 [==============================] - 3s 32ms/step - loss: 0.0976 - accuracy: 0.9982
Epoch 80/100
107/107 [==============================] - 3s 31ms/step - loss: 0.0960 - accuracy: 0.9982
Epoch 81/100
107/107 [==============================] - 3s 31ms/step - loss: 0.0944 - accuracy: 0.9982
Epoch 82/100
107/107 [==============================] - 3s 31ms/step - loss: 0.0928 - accuracy: 0.9982
Epoch 83/100
107/107 [==============================] - 3s 29ms/step - loss: 0.0913 - accuracy: 0.9982
Epoch 84/100
107/107 [==============================] - 3s 29ms/step - loss: 0.0898 - accuracy: 0.9982
Epoch 85/100
107/107 [==============================] - 3s 29ms/step - loss: 0.0883 - accuracy: 0.9982
Epoch 86/100
107/107 [==============================] - 3s 29ms/step - loss: 0.0869 - accuracy: 0.9982
Epoch 87/100
107/107 [==============================] - 3s 28ms/step - loss: 0.0856 - accuracy: 0.9982
Epoch 88/100
107/107 [==============================] - 3s 29ms/step - loss: 0.0842 - 

107/107 [==============================] - 3s 32ms/step - loss: 0.3923 - accuracy: 0.9983
Epoch 58/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3880 - accuracy: 0.9983
Epoch 59/100
107/107 [==============================] - 4s 33ms/step - loss: 0.3837 - accuracy: 0.9983
Epoch 60/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3795 - accuracy: 0.9983
Epoch 61/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3753 - accuracy: 0.9983
Epoch 62/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3712 - accuracy: 0.9983
Epoch 63/100
107/107 [==============================] - 4s 35ms/step - loss: 0.3671 - accuracy: 0.9983
Epoch 64/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3631 - accuracy: 0.9983
Epoch 65/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3593 - accuracy: 0.9983
Epoch 66/100
107/107 [==============================] - 4s 34ms/step - loss: 0.3554 - 

Epoch 37/50
64/64 [==============================] - 3s 45ms/step - loss: 0.3684 - accuracy: 0.9983
Epoch 38/50
64/64 [==============================] - 3s 44ms/step - loss: 0.3614 - accuracy: 0.9983
Epoch 39/50
64/64 [==============================] - 3s 44ms/step - loss: 0.3546 - accuracy: 0.9983
Epoch 40/50
64/64 [==============================] - 3s 44ms/step - loss: 0.3480 - accuracy: 0.9983
Epoch 41/50
64/64 [==============================] - 3s 45ms/step - loss: 0.3414 - accuracy: 0.9983
Epoch 42/50
64/64 [==============================] - 3s 44ms/step - loss: 0.3350 - accuracy: 0.9983
Epoch 43/50
64/64 [==============================] - 3s 45ms/step - loss: 0.3287 - accuracy: 0.9983
Epoch 44/50
64/64 [==============================] - 3s 45ms/step - loss: 0.3225 - accuracy: 0.9983
Epoch 45/50
64/64 [==============================] - 3s 45ms/step - loss: 0.3164 - accuracy: 0.9983
Epoch 46/50
64/64 [==============================] - 3s 51ms/step - loss: 0.3105 - accuracy: 0.9983


64/64 [==============================] - 3s 44ms/step - loss: 0.5590 - accuracy: 0.9982
Epoch 18/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5537 - accuracy: 0.9982
Epoch 19/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5484 - accuracy: 0.9982
Epoch 20/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5432 - accuracy: 0.9982
Epoch 21/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5382 - accuracy: 0.9982
Epoch 22/50
64/64 [==============================] - 3s 45ms/step - loss: 0.5333 - accuracy: 0.9982
Epoch 23/50
64/64 [==============================] - 3s 45ms/step - loss: 0.5285 - accuracy: 0.9982
Epoch 24/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5237 - accuracy: 0.9982
Epoch 25/50
64/64 [==============================] - 3s 44ms/step - loss: 0.5191 - accuracy: 0.9982
Epoch 26/50
64/64 [==============================] - 3s 45ms/step - loss: 0.5145 - accuracy: 0.9982
Epoch 27/50


64/64 [==============================] - 3s 45ms/step - loss: 0.4063 - accuracy: 0.9982
Epoch 49/50
64/64 [==============================] - 3s 46ms/step - loss: 0.4024 - accuracy: 0.9982
Epoch 50/50
16/16 [==============================] - 0s 6ms/step - loss: 0.3966 - accuracy: 0.9984
Epoch 1/50
64/64 [==============================] - 3s 45ms/step - loss: 13.3674 - accuracy: 0.9842
Epoch 2/50
64/64 [==============================] - 3s 45ms/step - loss: 0.6670 - accuracy: 0.9983
Epoch 3/50
64/64 [==============================] - 3s 45ms/step - loss: 0.6536 - accuracy: 0.9983
Epoch 4/50
64/64 [==============================] - 3s 46ms/step - loss: 0.6416 - accuracy: 0.9983
Epoch 5/50
64/64 [==============================] - 3s 45ms/step - loss: 0.6304 - accuracy: 0.9983
Epoch 6/50
64/64 [==============================] - 3s 46ms/step - loss: 0.6199 - accuracy: 0.9983
Epoch 7/50
64/64 [==============================] - 3s 46ms/step - loss: 0.6098 - accuracy: 0.9983
Epoch 8/50
64/64 [=

64/64 [==============================] - 3s 44ms/step - loss: 0.3657 - accuracy: 0.9983
Epoch 29/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3562 - accuracy: 0.9983
Epoch 30/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3470 - accuracy: 0.9983
Epoch 31/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3379 - accuracy: 0.9983
Epoch 32/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3291 - accuracy: 0.9983
Epoch 33/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3205 - accuracy: 0.9983
Epoch 34/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3121 - accuracy: 0.9983
Epoch 35/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3039 - accuracy: 0.9983
Epoch 36/100
64/64 [==============================] - 3s 44ms/step - loss: 0.2959 - accuracy: 0.9983
Epoch 37/100
64/64 [==============================] - 3s 44ms/step - loss: 0.2882 - accuracy: 0.9983
Epo

64/64 [==============================] - 3s 51ms/step - loss: 0.6210 - accuracy: 0.9983
Epoch 9/100
64/64 [==============================] - 3s 52ms/step - loss: 0.6144 - accuracy: 0.9983
Epoch 10/100
64/64 [==============================] - 3s 52ms/step - loss: 0.6080 - accuracy: 0.9983
Epoch 11/100
64/64 [==============================] - 3s 52ms/step - loss: 0.6019 - accuracy: 0.9983
Epoch 12/100
64/64 [==============================] - 3s 51ms/step - loss: 0.5960 - accuracy: 0.9983
Epoch 13/100
64/64 [==============================] - 3s 52ms/step - loss: 0.5902 - accuracy: 0.9983
Epoch 14/100
64/64 [==============================] - 3s 53ms/step - loss: 0.5847 - accuracy: 0.9983
Epoch 15/100
64/64 [==============================] - 3s 53ms/step - loss: 0.5793 - accuracy: 0.9983
Epoch 16/100
64/64 [==============================] - 3s 53ms/step - loss: 0.5741 - accuracy: 0.9983
Epoch 17/100
64/64 [==============================] - 3s 53ms/step - loss: 0.5690 - accuracy: 0.9983
Epoc

64/64 [==============================] - 3s 50ms/step - loss: 0.3365 - accuracy: 0.9983
Epoch 90/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3344 - accuracy: 0.9983
Epoch 91/100
64/64 [==============================] - 3s 51ms/step - loss: 0.3323 - accuracy: 0.9983
Epoch 92/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3302 - accuracy: 0.9983
Epoch 93/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3282 - accuracy: 0.9983
Epoch 94/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3261 - accuracy: 0.9983
Epoch 95/100
64/64 [==============================] - 3s 51ms/step - loss: 0.3241 - accuracy: 0.9983
Epoch 96/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3221 - accuracy: 0.9983
Epoch 97/100
64/64 [==============================] - 3s 50ms/step - loss: 0.3201 - accuracy: 0.9983
Epoch 98/100
64/64 [==============================] - 3s 52ms/step - loss: 0.3181 - accuracy: 0.9983
Epo

64/64 [==============================] - 3s 44ms/step - loss: 0.1984 - accuracy: 0.9982
Epoch 70/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1951 - accuracy: 0.9982
Epoch 71/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1919 - accuracy: 0.9982
Epoch 72/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1887 - accuracy: 0.9982
Epoch 73/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1856 - accuracy: 0.9982
Epoch 74/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1826 - accuracy: 0.9982
Epoch 75/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1796 - accuracy: 0.9982
Epoch 76/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1768 - accuracy: 0.9982
Epoch 77/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1739 - accuracy: 0.9982
Epoch 78/100
64/64 [==============================] - 3s 44ms/step - loss: 0.1712 - accuracy: 0.9982
Epo

64/64 [==============================] - 3s 44ms/step - loss: 0.5335 - accuracy: 0.9982
Epoch 50/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5299 - accuracy: 0.9982
Epoch 51/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5263 - accuracy: 0.9982
Epoch 52/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5227 - accuracy: 0.9982
Epoch 53/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5192 - accuracy: 0.9982
Epoch 54/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5156 - accuracy: 0.9982
Epoch 55/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5120 - accuracy: 0.9982
Epoch 56/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5085 - accuracy: 0.9982
Epoch 57/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5050 - accuracy: 0.9982
Epoch 58/100
64/64 [==============================] - 3s 44ms/step - loss: 0.5015 - accuracy: 0.9982
Epo

64/64 [==============================] - 3s 44ms/step - loss: 0.3848 - accuracy: 0.9983
Epoch 30/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3756 - accuracy: 0.9983
Epoch 31/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3666 - accuracy: 0.9983
Epoch 32/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3577 - accuracy: 0.9983
Epoch 33/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3491 - accuracy: 0.9983
Epoch 34/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3406 - accuracy: 0.9983
Epoch 35/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3322 - accuracy: 0.9983
Epoch 36/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3241 - accuracy: 0.9983
Epoch 37/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3162 - accuracy: 0.9983
Epoch 38/100
64/64 [==============================] - 3s 44ms/step - loss: 0.3085 - accuracy: 0.9983
Epo

54/54 [==============================] - 3s 53ms/step - loss: 0.6067 - accuracy: 0.9983
Epoch 10/50
54/54 [==============================] - 3s 53ms/step - loss: 0.5991 - accuracy: 0.9983
Epoch 11/50
54/54 [==============================] - 3s 53ms/step - loss: 0.5916 - accuracy: 0.9983
Epoch 12/50
54/54 [==============================] - 3s 53ms/step - loss: 0.5843 - accuracy: 0.9983
Epoch 13/50
54/54 [==============================] - 3s 54ms/step - loss: 0.5770 - accuracy: 0.9983
Epoch 14/50
54/54 [==============================] - 3s 54ms/step - loss: 0.5699 - accuracy: 0.9983
Epoch 15/50
54/54 [==============================] - 3s 61ms/step - loss: 0.5628 - accuracy: 0.9983
Epoch 16/50
54/54 [==============================] - 3s 60ms/step - loss: 0.5558 - accuracy: 0.9983
Epoch 17/50
54/54 [==============================] - 3s 58ms/step - loss: 0.5489 - accuracy: 0.9983
Epoch 18/50
54/54 [==============================] - 4s 67ms/step - loss: 0.5420 - accuracy: 0.9983
Epoch 19/50


54/54 [==============================] - 3s 59ms/step - loss: 0.3726 - accuracy: 0.9983
Epoch 41/50
54/54 [==============================] - 3s 61ms/step - loss: 0.3644 - accuracy: 0.9983
Epoch 42/50
54/54 [==============================] - 3s 60ms/step - loss: 0.3563 - accuracy: 0.9983
Epoch 43/50
54/54 [==============================] - 3s 63ms/step - loss: 0.3485 - accuracy: 0.9983
Epoch 44/50
54/54 [==============================] - 3s 55ms/step - loss: 0.3407 - accuracy: 0.9983
Epoch 45/50
54/54 [==============================] - 3s 53ms/step - loss: 0.3331 - accuracy: 0.9983
Epoch 46/50
54/54 [==============================] - 3s 53ms/step - loss: 0.3257 - accuracy: 0.9983
Epoch 47/50
54/54 [==============================] - 3s 53ms/step - loss: 0.3184 - accuracy: 0.9983
Epoch 48/50
54/54 [==============================] - 3s 52ms/step - loss: 0.3114 - accuracy: 0.9983
Epoch 49/50
54/54 [==============================] - 3s 53ms/step - loss: 0.3044 - accuracy: 0.9983
Epoch 50/50


54/54 [==============================] - 3s 53ms/step - loss: 0.4891 - accuracy: 0.9982
Epoch 21/50
54/54 [==============================] - 3s 53ms/step - loss: 0.4798 - accuracy: 0.9982
Epoch 22/50
54/54 [==============================] - 3s 54ms/step - loss: 0.4704 - accuracy: 0.9982
Epoch 23/50
54/54 [==============================] - 3s 54ms/step - loss: 0.4612 - accuracy: 0.9982
Epoch 24/50
54/54 [==============================] - 3s 61ms/step - loss: 0.4519 - accuracy: 0.9982
Epoch 25/50
54/54 [==============================] - 4s 65ms/step - loss: 0.4427 - accuracy: 0.9982
Epoch 26/50
54/54 [==============================] - 3s 60ms/step - loss: 0.4336 - accuracy: 0.9982
Epoch 27/50
54/54 [==============================] - 3s 62ms/step - loss: 0.4246 - accuracy: 0.9982
Epoch 28/50
54/54 [==============================] - 3s 61ms/step - loss: 0.4156 - accuracy: 0.9982
Epoch 29/50
54/54 [==============================] - 3s 64ms/step - loss: 0.4068 - accuracy: 0.9982
Epoch 30/50


14/14 [==============================] - 0s 6ms/step - loss: 0.2798 - accuracy: 0.9980
Epoch 1/100
54/54 [==============================] - 3s 53ms/step - loss: 42.2392 - accuracy: 0.9795
Epoch 2/100
54/54 [==============================] - 3s 54ms/step - loss: 0.6692 - accuracy: 0.9983
Epoch 3/100
54/54 [==============================] - 3s 54ms/step - loss: 0.6567 - accuracy: 0.9983
Epoch 4/100
54/54 [==============================] - 3s 54ms/step - loss: 0.6454 - accuracy: 0.9983
Epoch 5/100
54/54 [==============================] - 3s 53ms/step - loss: 0.6348 - accuracy: 0.9983
Epoch 6/100
54/54 [==============================] - 3s 54ms/step - loss: 0.6248 - accuracy: 0.9983
Epoch 7/100
54/54 [==============================] - 3s 53ms/step - loss: 0.6152 - accuracy: 0.9983
Epoch 8/100
54/54 [==============================] - 3s 60ms/step - loss: 0.6059 - accuracy: 0.9983
Epoch 9/100
54/54 [==============================] - 4s 65ms/step - loss: 0.5969 - accuracy: 0.9983
Epoch 10/100

54/54 [==============================] - 3s 58ms/step - loss: 0.1587 - accuracy: 0.9983
Epoch 82/100
54/54 [==============================] - 3s 58ms/step - loss: 0.1560 - accuracy: 0.9983
Epoch 83/100
54/54 [==============================] - 3s 56ms/step - loss: 0.1535 - accuracy: 0.9983
Epoch 84/100
54/54 [==============================] - 3s 56ms/step - loss: 0.1510 - accuracy: 0.9983
Epoch 85/100
54/54 [==============================] - 3s 57ms/step - loss: 0.1485 - accuracy: 0.9983
Epoch 86/100
54/54 [==============================] - 3s 56ms/step - loss: 0.1462 - accuracy: 0.9983
Epoch 87/100
54/54 [==============================] - 3s 56ms/step - loss: 0.1438 - accuracy: 0.9983
Epoch 88/100
54/54 [==============================] - 3s 57ms/step - loss: 0.1416 - accuracy: 0.9983
Epoch 89/100
54/54 [==============================] - 3s 57ms/step - loss: 0.1394 - accuracy: 0.9983
Epoch 90/100
54/54 [==============================] - 3s 61ms/step - loss: 0.1372 - accuracy: 0.9983
Epo

54/54 [==============================] - 4s 66ms/step - loss: 0.2265 - accuracy: 0.9983
Epoch 62/100
54/54 [==============================] - 4s 67ms/step - loss: 0.2223 - accuracy: 0.9983
Epoch 63/100
54/54 [==============================] - 4s 66ms/step - loss: 0.2182 - accuracy: 0.9983
Epoch 64/100
54/54 [==============================] - 4s 66ms/step - loss: 0.2143 - accuracy: 0.9983
Epoch 65/100
54/54 [==============================] - 3s 64ms/step - loss: 0.2104 - accuracy: 0.9983
Epoch 66/100
54/54 [==============================] - 4s 65ms/step - loss: 0.2066 - accuracy: 0.9983
Epoch 67/100
54/54 [==============================] - 3s 64ms/step - loss: 0.2028 - accuracy: 0.9983
Epoch 68/100
54/54 [==============================] - 3s 62ms/step - loss: 0.1992 - accuracy: 0.9983
Epoch 69/100
54/54 [==============================] - 3s 65ms/step - loss: 0.1957 - accuracy: 0.9983
Epoch 70/100
54/54 [==============================] - 4s 66ms/step - loss: 0.1923 - accuracy: 0.9983
Epo

54/54 [==============================] - 4s 67ms/step - loss: 0.3373 - accuracy: 0.9982
Epoch 42/100
54/54 [==============================] - 4s 67ms/step - loss: 0.3311 - accuracy: 0.9982
Epoch 43/100
54/54 [==============================] - 3s 63ms/step - loss: 0.3249 - accuracy: 0.9982
Epoch 44/100
54/54 [==============================] - 3s 58ms/step - loss: 0.3189 - accuracy: 0.9982
Epoch 45/100
54/54 [==============================] - 3s 58ms/step - loss: 0.3130 - accuracy: 0.9982
Epoch 46/100
54/54 [==============================] - 3s 57ms/step - loss: 0.3072 - accuracy: 0.9982
Epoch 47/100
54/54 [==============================] - 3s 56ms/step - loss: 0.3016 - accuracy: 0.9982
Epoch 48/100
54/54 [==============================] - 3s 56ms/step - loss: 0.2961 - accuracy: 0.9982
Epoch 49/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2906 - accuracy: 0.9982
Epoch 50/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2853 - accuracy: 0.9982
Epo

54/54 [==============================] - 3s 57ms/step - loss: 0.5204 - accuracy: 0.9982
Epoch 22/100
54/54 [==============================] - 3s 57ms/step - loss: 0.5137 - accuracy: 0.9982
Epoch 23/100
54/54 [==============================] - 3s 57ms/step - loss: 0.5069 - accuracy: 0.9982
Epoch 24/100
54/54 [==============================] - 3s 58ms/step - loss: 0.5002 - accuracy: 0.9982
Epoch 25/100
54/54 [==============================] - 3s 57ms/step - loss: 0.4935 - accuracy: 0.9982
Epoch 26/100
54/54 [==============================] - 3s 57ms/step - loss: 0.4869 - accuracy: 0.9982
Epoch 27/100
54/54 [==============================] - 3s 58ms/step - loss: 0.4804 - accuracy: 0.9982
Epoch 28/100
54/54 [==============================] - 3s 57ms/step - loss: 0.4739 - accuracy: 0.9982
Epoch 29/100
54/54 [==============================] - 3s 57ms/step - loss: 0.4674 - accuracy: 0.9982
Epoch 30/100
54/54 [==============================] - 3s 58ms/step - loss: 0.4610 - accuracy: 0.9982
Epo

54/54 [==============================] - 4s 68ms/step - loss: 1.0102 - accuracy: 0.9971
Epoch 2/100
54/54 [==============================] - 3s 64ms/step - loss: 0.6700 - accuracy: 0.9983
Epoch 3/100
54/54 [==============================] - 3s 65ms/step - loss: 0.6583 - accuracy: 0.9983
Epoch 4/100
54/54 [==============================] - 4s 66ms/step - loss: 0.6479 - accuracy: 0.9983
Epoch 5/100
54/54 [==============================] - 4s 65ms/step - loss: 0.6383 - accuracy: 0.9983
Epoch 6/100
54/54 [==============================] - 4s 65ms/step - loss: 0.6294 - accuracy: 0.9983
Epoch 7/100
54/54 [==============================] - 3s 62ms/step - loss: 0.6211 - accuracy: 0.9983
Epoch 8/100
54/54 [==============================] - 3s 59ms/step - loss: 0.6132 - accuracy: 0.9983
Epoch 9/100
54/54 [==============================] - 3s 59ms/step - loss: 0.6056 - accuracy: 0.9983
Epoch 10/100
54/54 [==============================] - 3s 58ms/step - loss: 0.5984 - accuracy: 0.9983
Epoch 11/10

54/54 [==============================] - 3s 57ms/step - loss: 0.2933 - accuracy: 0.9983
Epoch 83/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2906 - accuracy: 0.9983
Epoch 84/100
54/54 [==============================] - 3s 56ms/step - loss: 0.2880 - accuracy: 0.9983
Epoch 85/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2854 - accuracy: 0.9983
Epoch 86/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2828 - accuracy: 0.9983
Epoch 87/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2803 - accuracy: 0.9983
Epoch 88/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2778 - accuracy: 0.9983
Epoch 89/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2753 - accuracy: 0.9983
Epoch 90/100
54/54 [==============================] - 3s 57ms/step - loss: 0.2729 - accuracy: 0.9983
Epoch 91/100
54/54 [==============================] - 3s 58ms/step - loss: 0.2704 - accuracy: 0.9983
Epo

In [28]:
## Evaluating the model for the train and test dataset with best parameters
model_5.fit(x_train, y_train, epochs = grid_result.best_params_['epochs'], batch_size = grid_result.best_params_['batch_size'])

print('Training Accuracy of the new model after tuning the parameters : %.2f%%' %(100*model_5.score(x_train, y_train)))
print('')

model_5.fit(x_test, y_test, epochs = grid_result.best_params_['epochs'], batch_size = grid_result.best_params_['batch_size'])

print('Test Accuracy of the new model after tuning the parameters : %.2f%%' %(100*model_5.score(x_test, y_test)))
print('')

Epoch 1/50
133/133 [==============================] - 4s 30ms/step - loss: 14.1971 - accuracy: 0.9908
Epoch 2/50
133/133 [==============================] - 4s 30ms/step - loss: 0.6519 - accuracy: 0.9983
Epoch 3/50
133/133 [==============================] - 4s 30ms/step - loss: 0.6324 - accuracy: 0.9983
Epoch 4/50
133/133 [==============================] - 4s 29ms/step - loss: 0.6154 - accuracy: 0.9983
Epoch 5/50
133/133 [==============================] - 4s 29ms/step - loss: 0.6000 - accuracy: 0.9983
Epoch 6/50
133/133 [==============================] - 4s 29ms/step - loss: 0.5857 - accuracy: 0.9983
Epoch 7/50
133/133 [==============================] - 4s 29ms/step - loss: 0.5723 - accuracy: 0.9983
Epoch 8/50
133/133 [==============================] - 4s 29ms/step - loss: 0.5597 - accuracy: 0.9983
Epoch 9/50
133/133 [==============================] - 4s 29ms/step - loss: 0.5475 - accuracy: 0.9983
Epoch 10/50
133/133 [==============================] - 4s 29ms/step - loss: 0.5359 - accur

57/57 [==============================] - 2s 27ms/step - loss: 0.5690 - accuracy: 0.9983
Epoch 31/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5638 - accuracy: 0.9983
Epoch 32/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5586 - accuracy: 0.9983
Epoch 33/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5533 - accuracy: 0.9983
Epoch 34/50
57/57 [==============================] - 2s 28ms/step - loss: 0.5479 - accuracy: 0.9983
Epoch 35/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5426 - accuracy: 0.9983
Epoch 36/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5373 - accuracy: 0.9983
Epoch 37/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5319 - accuracy: 0.9983
Epoch 38/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5265 - accuracy: 0.9983
Epoch 39/50
57/57 [==============================] - 2s 27ms/step - loss: 0.5212 - accuracy: 0.9983
Epoch 40/50


## Obs : The result shows that the model is neither overfitting nor underfitting as both train and test accuracy is 99.83% which is preety good